storage/redis/peer_store.go

```diff
@@ -267,28 +238,77 @@ type peerStore struct {
 	wg     sync.WaitGroup
 }
 
-func (ps *peerStore) groups() []string {
-	return []string{bittorrent.IPv4.String(), bittorrent.IPv6.String()}
-}
+var addressFamilies = [2]string{"4", "6"}
 
-func (ps *peerStore) leecherInfohashKey(af, ih string) string {
-	return af + "_L_" + ih
-}
+// Key that stores the peers for an InfoHash.
+//
+// "4S0102030405060708090a0b0c0d0e0f1011121314" => IPv4 Seeder with Peer ID 0102030405060708090a0b0c0d0e0f1011121314
+// "6L0204090405060708060a0b0c0d0e0f1011121328" => IPv6 Seeder with Peer ID 0204090405060708060a0b0c0d0e0f1011121328
+func peerKey(infoHash bittorrent.InfoHash, seed bool, ip netaddr.IP) string {
+	var b strings.Builder
+	switch {
+	case ip.Is4(), ip.Is4in6():
+		b.WriteString("4")
+	case ip.Is6():
+		b.WriteString("6")
+	default:
+		panic("unknown IP version")
+	}
 
-func (ps *peerStore) seederInfohashKey(af, ih string) string {
-	return af + "_S_" + ih
-}
+	if seed {
+		b.WriteString("S")
+	} else {
+		b.WriteString("L")
+	}
 
-func (ps *peerStore) infohashCountKey(af string) string {
-	return af + "_infohash_count"
+	b.WriteString(infoHash.String())
```

Comment:
We could use `RawString` to save 20 bytes, which is nice, but makes the keys non-human readable. I think that's what we used before, semantically?

Reply:
Yep, I actually swapped these around and forgot to change this back. It's not supposed to be hex here.